src/resources/parser/glb-parser.js

```diff
@@ -1785,20 +1785,19 @@ const createLight = function (gltfLight, node) {
 
     // glTF stores light already in energy/area, but we need to provide the light with only the energy parameter,
     // so we need the intensities in candela back to lumen
-    switch (gltfLight.type) {
-        case "spot": {
-            const falloffEnd = Math.cos(gltfLight.spot.innerConeAngle);
-            const falloffStart = Math.cos(gltfLight.spot.outerConeAngle);
-            lightProps.luminance = gltfLight.intensity * (2 * Math.PI * ((1 - falloffStart) + (falloffStart - falloffEnd) / 2.0));
-            break;
+    if (gltfLight.hasOwnProperty("intensity")) {
+        switch (gltfLight.type) {
+            case "spot":
+                lightProps.luminance = gltfLight.intensity * Light.getLightUnitConversion(LIGHTTYPE_SPOT, gltfLight.spot.outerConeAngle, gltfLight.spot.innerConeAngle);
+                break;
+            case "point":
+                lightProps.luminance = gltfLight.intensity * Light.getLightUnitConversion(LIGHTTYPE_OMNI);
+                break;
+            case "directional":
+                // Directional light luminance is already in lux
+                lightProps.luminance = gltfLight.intensity;
+                break;
```

Comment:
I think this whole section could be as simple as this
```
lightProps.luminance = gltf.intensity * Light.getLightUnitConversion(lightProps.type, lightProps.innerConeAngle * math.DEG_TO_RAD, lightProps.outerConeAngle * math.DEG_TO_RAD);
```

Reply:
I wanted to avoid having to do the math with converting degrees to radians and providing those arguments, but since it's loader code it's not so vital I suppose.